include/nuttx/compiler.h

```diff
@@ -1110,6 +1110,18 @@
 
 #endif
 
+/* Green Hills Software definitions *****************************************/
+
+#if defined(__ghs__)
```

Comment:
I think this should be placed under `#ifdef __GNUC__` block above.

Reply:
Sorry, I missed https://github.com/apache/nuttx/pull/12886#discussion_r1713097499 when was posting a review.
Still I think that if greenhills compiler  defines `__GNUC__` the `#if defined(__ghs__)` should be encapsulated under `__GNUC__ ` branch and not at the end off all compilers selection

Reply:
ok